sycl/source/detail/scheduler/commands.cpp

```diff
@@ -1949,6 +1949,31 @@ cl_int ExecCGCommand::enqueueImp() {
 
     return CL_SUCCESS;
   }
+  case CG::CGTYPE::BARRIER: {
+    if (MQueue->get_device().is_host()) {
+      // NOP for host device.
+      return PI_SUCCESS;
+    }
+    const detail::plugin &Plugin = MQueue->getPlugin();
+    Plugin.call<PiApiKind::piEnqueueBarrier>(MQueue->getHandleRef());
```

Comment:
Just looked into other PI implementations, mb it is better to call 
```
Plugin.call<PiApiKind::piEnqueueBarrierWait>(
        MQueue->getHandleRef(), 0, nullptr, &Event);
```
here and remove ```piEnqueueBarrier``` from PI.
Such call is correct for OpenCL according to spec: https://www.khronos.org/registry/OpenCL/sdk/1.2/docs/man/xhtml/clEnqueueBarrierWithWaitList.html